papaparse.js

```diff
@@ -1111,32 +1110,38 @@ License: MIT
 			}
 
 			if (needsHeaderRow())
-				fillHeaderFields();
+				fillHeaderFields(isStep);
 
-			return applyHeaderAndDynamicTypingAndTransformation();
+			return applyHeaderAndDynamicTypingAndTransformation(isStep);
 		}
 
 		function needsHeaderRow()
 		{
 			return _config.header && _fields.length === 0;
 		}
 
-		function fillHeaderFields()
+		function fillHeaderFields(isStep)
 		{
 			if (!_results)
 				return;
-			for (var i = 0; needsHeaderRow() && i < _results.data.length; i++)
-				for (var j = 0; j < _results.data[i].length; j++)
-				{
-					var header = _results.data[i][j];
 
-					if (isFunction(_config.transformHeader)) {
-						header = _config.transformHeader(header);
-					}
+			function addHeder(header)
+			{
+				if (isFunction(_config.transformHeader))
+					header = _config.transformHeader(header);
 
-					_fields.push(header);
-				}
-			_results.data.splice(0, 1);
+				_fields.push(header);
+			}
+
+			if (isStep)
```

Comment:
can we test that _config.step is set instead of passing the isStep variable everywhere?

Reply:
will do.

Reply:
Okay. this is a problem. `_config.step` is truthy even when you are processing outside of a step...  the problem we are attempting to fix is that the step function calls on the process/parse functions that assume the shape of results is a two dimensional array, when for steps, it's just a single dimension.  We need to tell the those functions the context is one row. not a chunk of rows... Alternatively we could ensure steps use a two dimensional array as well, but limit the length to 1 row... but that is way more invasive at the moment.

Reply:
Then probably is better to make those functions always work with a one dimensional array.

Reply:
So, are you suggesting to test the shape of `_results.data` instead of passing in the flag?

Reply:
done. :)